src/app/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js

```diff
@@ -154,7 +159,7 @@ export class CheckoutOrderSummary extends PureComponent {
                     { this.renderDiscount() }
                     { this.renderPriceLine(tax_amount, __('Tax')) }
                     { checkoutStep !== SHIPPING_STEP
-                        ? this.renderPriceLine(grand_total + shipping_amount, __('Order total'))
+                        ? this.renderPriceLine(payment_grand_total || grand_total, __('Order total'))
```

Comment:
If I remember correctly, `paymentTotals` is what you have in the LocalStorage, whereas `totals` is the freshest information from the server. Wouldn't it make more sense to fall back from the most relevant information to the one cached in the browser? I propose swapping this || statement's operands.

Reply:
Yes, but the only reason why added back payment_grand_total as 1 prior since for some reason when checkout is updating for the for time shipping amount (null -> x $) in cart_totals grand_total is not updating (not adding shipping amount into grand) without cart update ( add/remove product or refresh page ). So that user will see grand_total without calc shipping amount. Will try to remake it atm and see what can be change to set freshest information from the server.

Reply:
hello @yeegor !
I have tried to fix issue with grand_total for checkoutTotals, but  unsuccessfully. Still issue appears when cart_totals shipping amount is changed for the first time and grand totals is not including it. Solution can be to update cart when checkout.container is setting data into paymentTotals and etc, but due to lack of time didn't managed to fix all issues to solve it. So at the moment this fix will works, but ofcourse better to use newest data as you mentioned.